### PR TITLE
Handle assets if public page + add favicons generator

### LINKED ISSUE #237: Better handling vendor/assets
Better location in files outline + handle when there is no `vendor/assets`

### PR DESCRIPTION
**Pull request by CPatchane**
⚠️ __BREAKING CHANGES__ ⚠️ 

* The targets assets are now in `targets/assets` instead of `targets/vendor/assets`
* Favicons are now generated and automatically injected into html files (except safari pinned icon)
* The `webpack.config.vendors` has been renamed to `webpack.config.assets`
* Assets are correctly handled if the application has public page

Also fix #237

### CODE REVIEW

File: packages/cozy-scripts/config/webpack.config.assets.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  plugins: [
+    {
+      removeViewBox: false,
+      // inlince styles are blocked by Cozy CSPs
+      inlineStyles: {
+        onlyMatchedOnce: false
+      }
+    }
+  ]
+})
```
Comment (Crash--) [inline packages/cozy-scripts/config/webpack.config.assets.js]: Do we really want svgo optimization? Can it breaks some svg?

File: packages/cozy-scripts/config/webpack.config.assets.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+const appIconPath = path.join(paths.appAssets(), manifest.icon)
+
+module.exports = {
+  plugins: [
+    new CopyPlugin([
+      {
+        from: appIconPath,
+        to: hasPublic
+          ? path.join(paths.appBuild(), publicFolderName, manifest.icon)
+          : path.join(paths.appBuild(), manifest.icon),
```
Comment (Crash--) [inline packages/cozy-scripts/config/webpack.config.assets.js]: Cool! With this, I can remove this file, right ? https://github.com/cozy/cozy-drive/blob/master/webpack/appicon.config.js

Comment (CPatchane) [reply packages/cozy-scripts/config/webpack.config.assets.js]: Yep! Exactly :)

**Comment by CPatchane** [comment]
@Crash-- What was the problem about the fonts? Path resolving on mobile?

**Pull request merged**